idea/idea-android/src/org/jetbrains/kotlin/android/quickfix/KotlinAndroidViewConstructorFix.kt

```diff
@@ -0,0 +1,126 @@
+ */
+
+package org.jetbrains.kotlin.android.quickfix
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.*
+import com.intellij.psi.util.PsiTypesUtil
+import org.jetbrains.android.facet.AndroidFacet
+import org.jetbrains.kotlin.asJava.toLightClass
+import org.jetbrains.kotlin.diagnostics.Diagnostic
+import org.jetbrains.kotlin.diagnostics.Errors.SUPERTYPE_NOT_INITIALIZED
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.core.ShortenReferences
+import org.jetbrains.kotlin.idea.core.replaced
+import org.jetbrains.kotlin.idea.quickfix.KotlinQuickFixAction
+import org.jetbrains.kotlin.idea.quickfix.KotlinSingleIntentionActionFactory
+import org.jetbrains.kotlin.idea.util.addAnnotation
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.containingClass
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.DescriptorToSourceUtils
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameUnsafe
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.types.typeUtil.supertypes
+
+class KotlinAndroidViewConstructorFix(element: KtSuperTypeEntry) : KotlinQuickFixAction<KtSuperTypeEntry>(element) {
+
+    override fun getText() = "Add Android View constructors using '@JvmOverloads'"
+    override fun getFamilyName() = text
+
+    override fun isAvailable(project: Project, editor: Editor?, file: PsiFile): Boolean {
+        if (AndroidFacet.getInstance(file) == null) return false
+        return super.isAvailable(project, editor, file)
+    }
+
+    override fun invoke(project: Project, editor: Editor?, file: KtFile) {
+        val element = element ?: return
+        val ktClass = element.containingClass() ?: return
+
+        val factory = KtPsiFactory(element)
+
+        val newPrimaryConstructor = factory.createDeclarationByPattern<KtClass>(
+                "class A constructor(\n $0, $1, $2\n)",
+                factory.createParameter("context: android.content.Context"),
+                factory.createParameter("attrs: android.util.AttributeSet? = null"),
+                factory.createParameter("defStyleAttr: Int = 0")
+        ).primaryConstructor!!
+
+        val primaryConstructor = ktClass.createPrimaryConstructorIfAbsent().replaced(newPrimaryConstructor)
+        ShortenReferences.DEFAULT.process(primaryConstructor.valueParameterList!!)
+
+        primaryConstructor.addAnnotation(fqNameAnnotation, whiteSpaceText = " ")
+
+        element.replace(factory.createSuperTypeCallEntry(element.text + "(context, attrs, defStyleAttr)"))
+    }
+
+    companion object Factory : KotlinSingleIntentionActionFactory() {
+
+        private val fqNameAnnotation = FqName("kotlin.jvm.JvmOverloads")
+
+        private val requireConstructorParameterTypes = listOf(
```

Comment:
I think we may check only for presence third one `"android.content.Context", "android.util.AttributeSet", "int"`, since we using only it, if i understood correctly